### DISCUSSION TITLE
Pull rtmp source

### DISCUSSION THREAD

**Discussion by veramarvin** [OvenMediaLabs/OvenMediaEngine#1081] [Ideas]
Add a feature that allows pulling from rtmp source for restreaming

**Comment by dimiden** [comment]
Hi, @veramarvin

We have a plan to implement this feature.
The feature is not just restreaming, it's mainly used to make sure that the stream is being ingested properly. So, I think it is a feature that must be added later.

Thank you for your suggestion.

**Comment by AskAlice** [comment]
+1 on this. total blocker on using this library if I can't restream. would require another transcode hop via [datarhei/restreamer](https://github.com/datarhei/restreamer)

**Comment by mpisat** [comment]
+1 on this. If we can pull streams via RTMP when the client makes a request, it would be awesome. I really hate placing FFmpeg between things. If we can also pass through parameters like ...stream.key?some=1&params=2 that would be even better.

**Comment by tabakhian** [comment]
any update on this? any one have an idea how to do it manually?

**Comment by dimiden** [comment]
Nothing has been updated yet.
We don't have time to implement the RTMP Pull feature right now while developing other features.

**Comment by DufeRob** [comment]
Hello. I want to try ovenStream. But I need to pull the RTMP stream. Is it not possible to restream yet?

**Comment by IanMitchell77** [comment]
@AskAlice - you can pull and push the RTMP stream?

@tabakhian - you asked how to do it, did you find out? I do it with ffmpeg if thats any help?

**Comment by tabakhian** [comment]
unfortunately, I couldn't use OvenEngine, because of this problem, I had to
use OSSRS to have both rtmp and webrtc playing at the same time.

**Comment by getroot** [comment]
I know there are a lot of people who want this feature, and I found it to be very helpful in integrating with nginx.
So I will increase the priority of this feature. But we are so busy these days that we could start implementing this feature in early 2022.

**Comment by mpisat** [comment]
How does nginx integration for rtmp pull works? Is there any sample? I don't think nginx supports rtsp?

**Comment by getroot** [comment]
@mpisat 

Sorry but I don't know if I understood your question correctly.

I found this useful in a structure like below, and I got the idea here: https://github.com/AirenSoft/OvenMediaEngine/issues/536

NginX -> (rtmp pull) <- {OriginMap}OvenMediaEngine -> (WebRTC)

**Comment by getroot** [comment]
As we had anticipated the end of RTMP, any issues related to it had a very low priority for us. However, it seems that the lifespan of RTMP has been extended due to #1167. Consequently, we have started to reconsider features related to RTMP.
